src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -417,7 +417,7 @@ public void Destroy()
 				_fragmentManager
 					.BeginTransaction()
 					.Remove(_modalFragment)
-					.Commit();
+					.CommitAllowingStateLossEx();
```

Comment:
Can we try using the code here?

https://github.com/dotnet/maui/blob/2a9f9fd1980d521d1a36f72febc5ad9f94dc706e/src/Core/src/Platform/Android/Navigation/NavigationRootManager.cs#L139-L153

Reply:
@PureWeen just to make sure i'm understanding you correctly. You mean to update this block from 417 to 420 to use the same pattern you provided?

Reply:
Updated in https://github.com/dotnet/maui/pull/23400/commits/a1956060c6f3c3bc6cea98322b86c848a9282816

Reply:
I excluded 

```
if (_modalFragment is null) return;
```

and 

```
_modalFragment = null;
```

Because _modalFragment isn't nullable in this context.						
Happy to make that change if that will cause problems with garbage collection.

Reply:
Looks great!

If you want to try your hand at adding a UITest here are some good samples

https://github.com/dotnet/maui/blob/776dad91553c96ead0622da41b4c1ea8325b9fd0/src/Controls/tests/TestCases.HostApp/Issues/Issue11501.cs#L14
https://github.com/dotnet/maui/blob/776dad91553c96ead0622da41b4c1ea8325b9fd0/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue11501.cs#L8

here's some setup info
https://github.com/dotnet/maui/wiki/UITests

Reply:
@PureWeen thanks! I'll dive in and see what i can do. Thanks for your help!

Reply:
@PureWeen updated the pull request with some tests.

We were only able to replicate the issue by increasing the AnimationDuration on line https://github.com/lampo/maui/blob/a1956060c6f3c3bc6cea98322b86c848a9282816/src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs#L91. In our case we got into a failing state before the code change by increasing it to 1500.

This leads me to believe that the issue is occurring on slower hardware. The change does address this issue and it works regardless of animation duration.